src/main/java/executor/service/service/parallelflowexecutor/impls/subscribers/ExecutableScenarioComposerImpl.java

```diff
@@ -0,0 +1,90 @@
+package executor.service.service.parallelflowexecutor.impls.subscribers;
+
+import executor.service.model.proxy.ProxyConfigHolder;
+import executor.service.model.scenario.Scenario;
+import executor.service.model.service.ExecutableScenario;
+import executor.service.queue.ProxyQueue;
+import executor.service.queue.ScenarioQueue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExecutableScenarioComposerImpl implements ExecutableScenarioComposer {
+
+    private static final Logger log = LoggerFactory.getLogger(ExecutableScenarioComposerImpl.class);
+
+    private final ProxyQueue proxyQueue;
+    private final ScenarioQueue scenarioQueue;
+    private final ProxyConfigHolder defaultProxy;
+
+    public ExecutableScenarioComposerImpl(ProxyQueue proxyQueue, ScenarioQueue scenarioQueue, ProxyConfigHolder defaultProxy) {
+        this.proxyQueue = proxyQueue;
+        this.scenarioQueue = scenarioQueue;
+        this.defaultProxy = defaultProxy;
+    }
+
+    @Override
+    public List<ExecutableScenario> composeExecutableScenarios() {
+        try {
+            List<ProxyConfigHolder> proxies = proxyQueue.getAllProxy();
+            List<Scenario> scenarios = scenarioQueue.getAllScenario();
+            if (!scenarios.isEmpty() && !proxies.isEmpty()) {
+                return composeNotEmptyList(proxies, scenarios);
+            } else if (!scenarios.isEmpty()) {
+                return composeByDefaultProxy(defaultProxy, scenarios);
+            } else {
+                proxies.forEach(proxyQueue::addProxy);
+            }
+        } catch (InterruptedException e) {
+            log.error(e.getMessage());
```

Comment:
Add more details to facilitate debugging and performance analysis.

Reply:
> Add more details to facilitate debugging and performance analysis.

Are you sure that is necessary to get all print stack on catching exception?

Reply:
> I suggest customizing the message more clearly for debugging, I'm not talking about stack on catching exception.

If you need to debug this class, it's better add your message, but for error, it's quiet clear to show class where excpetion logged and reason of exception, I don't know what details and reasons to debug you need.